packages/protocol-kit/src/adapters/web3/contracts/SignMessageLib/v1.4.1/SignMessageLibContract_V1_4_1_Web3.ts

```diff
@@ -1,10 +1,119 @@
+  SignMessageFunction
+} from '@safe-global/protocol-kit/contracts/AbiType/SignMessageLib/SignMessageLibBaseContract'
 
-class SignMessageLibContract_V1_4_1_Web3 extends SignMessageLibWeb3Contract {
-  constructor(public contract: SignMessageLib) {
-    super(contract)
+// Remove all nested `readonly` modifiers from the ABI type
+type SignMessageLibContract_v1_4_1_Abi = DeepWriteable<SignMessageLibContract_v1_4_1_Abi_Readonly>
+
+/**
+ * SignMessageLibContract_v1_4_1_Web3 is the implementation specific to the SignMessageLib contract version 1.4.1.
+ *
+ * This class specializes in handling interactions with the SignMessageLib contract version 1.4.1 using Web3.js v6.
+ *
+ * @extends SignMessageLibBaseContractWeb3<SignMessageLibContract_v1_4_1_Abi> - Inherits from SignMessageLibBaseContractWeb3 with ABI specific to SignMessageLib contract version 1.4.1.
+ * @implements SignMessageLibContract_v1_4_1_Contract - Implements the interface specific to SignMessageLib contract version 1.4.1.
+ */
+class SignMessageLibContract_v1_4_1_Web3
+  extends SignMessageLibBaseContractWeb3<SignMessageLibContract_v1_4_1_Abi>
+  implements SignMessageLibContract_v1_4_1_Contract
+{
+  safeVersion: SafeVersion
+
+  /**
+   * Constructs an instance of SignMessageLibContract_v1_4_1_Web3
+   *
+   * @param chainId - The chain ID where the contract resides.
+   * @param web3Adapter - An instance of Web3Adapter.
+   * @param customContractAddress - Optional custom address for the contract. If not provided, the address is derived from the SignMessageLib deployments based on the chainId and safeVersion.
+   * @param customContractAbi - Optional custom ABI for the contract. If not provided, the default ABI for version 1.4.1 is used.
+   */
+  constructor(
+    chainId: bigint,
+    web3Adapter: Web3Adapter,
+    customContractAddress?: string,
+    customContractAbi?: SignMessageLibContract_v1_4_1_Abi
+  ) {
+    const safeVersion = '1.4.1'
+    const defaultAbi =
+      signMessageLib_1_4_1_ContractArtifacts.abi as DeepWriteable<SignMessageLibContract_v1_4_1_Abi>
+
+    super(chainId, web3Adapter, defaultAbi, safeVersion, customContractAddress, customContractAbi)
+
+    this.safeVersion = safeVersion
+  }
+
+  encode: EncodeSignMessageLibFunction<SignMessageLibContract_v1_4_1_Abi_Readonly> = (
+    functionToEncode,
+    args
+  ) => {
+    return this.contract.methods[functionToEncode](...args).encodeABI()
+  }
+
+  estimateGas: EstimateGasSignMessageLibFunction<
+    SignMessageLibContract_v1_4_1_Abi_Readonly,
+    Web3TransactionOptions
+  > = (functionToEstimate, args, options = {}) => {
+    return this.contract.methods[functionToEstimate](...args)
+      .estimateGas(options)
+      .then(BigInt)
+  }
+
+  getAddress: GetAddressSignMessageLibFunction = () => {
+    return Promise.resolve(this.contract.options.address)
+  }
+
+  async getMessageHash(args: readonly [string]): Promise<readonly [string]> {
+    return [await this.contract.methods.getMessageHash(...args)]
```

Comment:
Should this not be here and also for the 1.3.0 implementation?
```suggestion
    return [await this.contract.methods.getMessageHash(...args).call()]
```

Reply:
true!